src/Tools/dotnet-dsrouter/USBMuxTcpClientRouterFactory.cs

```diff
@@ -0,0 +1,492 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Diagnostics.Tools.DiagnosticsServerRouter
+{
+    internal class USBMuxInterop
+    {
+        public const string CoreFoundationLibrary = "/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation";
+        public const string MobileDeviceLibrary = "/System/Library/PrivateFrameworks/MobileDevice.framework/MobileDevice";
+        public const string LibC = "libc";
```

Comment:
How stable are these locations? Especially between 10.15 and 11 (and soon 12).

Reply:
The CoreFoundationLibrary should be stable, referenced from a lot of places, like https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Common/src/Interop/OSX/Interop.Libraries.cs#L8.

The libc is looked up using system loader, if that fails there will already been issues with runtime, so we won't get to this point unless that works.

The last one MobileDeviceLibrary is the one that could move around (since its a private framework), but there are code elsewhere that have it referenced like this and if we fail to load that library we will fallback (with error), so there will be software broken (not just this) when/if this gets moved or even renamed.